packages/data-context/src/sources/migration/utils.ts

```diff
@@ -7,3 +12,53 @@ export const isDefaultSupportFile = (supportFile: string) => {
 
   return false
 }
+
+export async function getDefaultSpecFileName (
+  { currentProject, testingType, fileExtensionToUse, specPattern, specs = [], name }:
+  { currentProject: string | null, testingType: TestingType | null, fileExtensionToUse: 'js' | 'ts', specPattern: string[], specs?: FoundSpec[], name?: string },
+): Promise<string> {
+  const debug = Debug('cypress:data-context:sources:migration:utils')
+
+  const defaultFilename = `${name ? name : testingType === 'e2e' ? 'spec' : 'ComponentName'}.cy.${fileExtensionToUse}`
+  const defaultPathname = path.join('cypress', testingType ?? 'e2e', defaultFilename)
+
+  if (!currentProject || !testingType) {
```

Comment:
Another more concise way that seems more in line with TypeScript's recommendations is an [assert function](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions). Example 

https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/actions/ProjectActions.ts#L159-L160

```ts
import assert from 'assert'

assert(currentProject, 'currentProject is required')
```

etc.

That said, do we even need this? Instead of throwing and returning here, should we just do

```ts
if (!currentProject || !testingType) {
    debug('currentProject or testingType undefined. Error intelligently detecting default filename, using safe default %o', err)
    return defaultPathname
}
```